pkg/config/creds.go

```diff
@@ -111,3 +122,51 @@ func (cm *CredentialManager) getCredsFromCache() (Creds, bool) {
 
 	return Creds{}, false
 }
+
+func (cm *CredentialManager) refresh(logger logr.Logger) error {
+	oldCreds := cm.creds
+	cm.creds = Creds{}
+
+	newCreds, err := cm.GetCredentials()
+
+	if err != nil {
+		return err
+	}
+
+	if oldCreds != newCreds {
+		logger.Info("Credentials have changed, updating creds")
+		// callbacks
+		cm.creds = newCreds
```

Comment:
this line can be removed, as `cacheCreds` called by `getCredentials` in line 130 will already set `cm.creds`
```suggestion
```